airflow/dag_processing/bundles/git.py

```diff
@@ -0,0 +1,114 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING
+
+from git import Repo
```

Comment:
Airflow core does not currently depend on `git`. Tests don’t fail since it’s a dev dependency (IIRC).

Reply:
I thought of this late night at dinner and then forgot. Thanks!

I also was considering whether we wanted this in core or a provider. But we don't have a natural home for it today. Something to consider longer term imo.

Reply:
Done. Going to leave this unresolved for others to potentially chime in though.

Reply:
Maybe we should have only the local bundle in the core while `git` bundle as provider.

Reply:
Yeah, I imagine we will likely end up with a git provider before 3. Going to defer that for now though.